ext/json/ext/parser/name_cache.h

```diff
@@ -0,0 +1,74 @@
+#ifndef _NAME_CACHE_H_
+#define _NAME_CACHE_H_
+
+#include "ruby.h"
+#include <string.h>
+
+#ifndef RB_UNLIKELY
+#define RB_UNLIKELY(expr) expr
+#endif
+
+#define JSON_RSTRING_CACHE_CAPA 63
+#define JSON_RSTRING_CACHE_MAX_ENTRY_LENGTH 55
+
+typedef struct rstring_cache_struct {
+    int length;
+    VALUE entries[JSON_RSTRING_CACHE_CAPA];
+} rstring_cache;
+
+static inline int rstring_cache_cmp(const char *str, const long length, VALUE rstring)
+{
+    long rstring_length = RSTRING_LEN(rstring);
+    if (length == rstring_length) {
+        return memcmp(str, RSTRING_PTR(rstring), length);
+    } else {
+        return (int)(length - rstring_length);
+    }
+}
+
+static void rstring_cache_insert_at(rstring_cache *cache, int index, VALUE rstring)
+{
+    MEMMOVE(&cache->entries[index + 1], &cache->entries[index], VALUE, cache->length - index);
+    cache->length ++;
+    cache->entries[index] = rstring;
+}
+
+static VALUE rstring_cache_fetch(rstring_cache *cache, const char *str, const long length)
+{
+    if (RB_UNLIKELY(length > JSON_RSTRING_CACHE_MAX_ENTRY_LENGTH)) {
+        rb_enc_interned_str(str, length, rb_utf8_encoding());
```

Comment:
Is this missing a `return`?
Fixed in the next commit, but I wonder if it affects the benchmark results for this commit.

Reply:
Yeah, it was a mistake. I don't think we ever hit that branch on any of the benchmarks.